src/libs/keystore/keystore.ts

```diff
@@ -179,6 +185,12 @@ export class Keystore {
     })
     // Persist the new secrets
     await this.storage.set('keystoreSecrets', secrets)
+
+    // produce uid if not exist (should created when first secret added)
+    if (!(await this.storage.get('uid', null))) {
+      const uid = keccak256(randomBytes(32)).slice(2, 34)
```

Comment:
why not just create this when the first secret is actually added (not unlocked) and why not have it be a hash of the mainKey